### PR TITLE
SB-25552 - UCI API's(federation-service) Deployment

### PR DESCRIPTION
**Pull request by shiva-rakshith**
<table>
<td>
  <a href="https://project-sunbird.atlassian.net/browse/SB-25552" title="SB-25552" target="_blank">StorySB-25552</a>  UCI - Deployment
  </td></table>
  <br />
 

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Enhancement (additive changes to improve performance)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by shiva-rakshith** [comment]
@keshavprasadms Please review the PR.

File: kubernetes/helm_charts/core/uci/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+appVersion: "1.0"
+description: A Helm chart for Kubernetes
+name: uciapi
```
Comment (keshavprasadms) [inline kubernetes/helm_charts/core/uci/Chart.yaml:4]: Why dont we use the name as `uci`. And then in confirmap references, we can use `{{ .Chart.Name }}-config`

File: ansible/roles/stack-sunbird/templates/uci_api.env [hunk 2]
```diff
@@ -0,0 +1,19 @@
+ENV=prod
+DATABASE_DEBUG=true
+HASURA_GRAPHQL_ADMIN_SECRET=HaTUNsNSaU
```
Comment (keshavprasadms) [inline ansible/roles/stack-sunbird/templates/uci_api.env:3]: No default password please

File: ansible/roles/stack-sunbird/templates/uci_api.env [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+KAFKA_USER=""
+POSTGRES_DB={{uci_bot_postgres_database}}
+POSTGRES_PASSWORD={{uci_postgres_password}}
+POSTGRES_USER={{uci_postgres_user}}
+PSQL_DB_URL_DEV=postgres://{{uci_postgres_user}}:{{uci_postgres_password}}@{{uci_postgres_host}}:5432/{{uci_bot_postgres_database}}
+REDIS_HOST={{sunbird_redis_host}}
+REDIS_PASS=""
+REDIS_PORT=6379
+UCI_CORE_BASE_URL=11.2.6.6/inbound-service
+
```
Comment (keshavprasadms) [inline ansible/roles/stack-sunbird/templates/uci_api.env:19]: Remove newlines

File: kubernetes/helm_charts/core/uci/Chart.yaml (hunk 1 above)
Comment (shiva-rakshith) [reply kubernetes/helm_charts/core/uci/Chart.yaml]: Changed the name to uci.

File: ansible/roles/stack-sunbird/templates/uci_api.env (hunk 2 above)
Comment (shiva-rakshith) [reply ansible/roles/stack-sunbird/templates/uci_api.env]: Changed to HASURA_GRAPHQL_ADMIN_SECRET={{hasura_graphql_admin_secret}}, as discussed in #2702

File: ansible/roles/stack-sunbird/templates/uci_api.env (hunk 3 above)
Comment (shiva-rakshith) [reply ansible/roles/stack-sunbird/templates/uci_api.env]: Removed new lines.

**Comment by keshavprasadms** [comment]
Please resolve conflicts

**Comment by shiva-rakshith** [comment]
@keshavprasadms Resolved conflicts.

**Review by keshavprasadms — APPROVED** [review]

**Pull request merged**